### PR TITLE
Refactor test suite

### PR DESCRIPTION
**Pull request by gpotter2**
- ~~on top https://github.com/ivre/masscanned/pull/41~~
- adds a functional test for SMB
- full refactor of the test suite & cleanups
image
- ability to select what tests to run: `TESTS=smb ./test`
image

### CODE REVIEW

File: test/src/tests/smb.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            "--option=client min protocol=NT1",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        text=True,
+    )
+    out, _ = proc.communicate()
+    assert f"Connecting to {IPV4_ADDR} at port 445" in out, "\n" + out
+    assert "session request ok" in out, "\n" + out
+    assert f"negotiated dialect[NT1] against server[{IPV4_ADDR}]" in out, "\n" + out
```
Comment (gpotter2) [inline test/src/tests/smb.py:43]: The functional test

File: test/test_masscanned.py [hunk 2]
```diff
@@ -172,7 +160,7 @@ def setup_net(iface):
 masscanned = subprocess.Popen(
     [
         "./target/debug/masscanned",
-        "-vvvvv",
+        "-v",
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply test/test_masscanned.py:163]: Sorry this is a debug leftover

**Review by Frky — COMMENTED** [review]
Thank you for refactoring tests.
Could you split ip46 into several files (`TCP`, `UDP`, and maybe `IP`?)?

**Comment by Frky** [comment]
Yes indeed, you are right. My bad. Although I think `TCP` tests deserve to be separated in a specific file.

**Comment by gpotter2** [comment]
All passing but the nmap test

**Pull request merged**